src/ray/raylet/local_task_manager.cc

```diff
@@ -491,15 +491,16 @@ void LocalTaskManager::SpillWaitingTasks() {
 }
 
 bool LocalTaskManager::TrySpillback(const std::shared_ptr<internal::Work> &work,
+                                    const TaskSpecification &spec,
                                     bool &is_infeasible) {
   auto scheduling_node_id = cluster_resource_scheduler_->GetBestSchedulableNode(
-      work->task.GetTaskSpecification(),
```

Comment:
Task specification creation is pretty expensive, which requires to compute resource at construction.
https://github.com/ray-project/ray/blob/e393a716d8742a987a36df555defb2ca90bb94d4/src/ray/common/task/task_spec.cc#L91-L131

Reply:
`work->task` has a cached TaskSpecification so it doesn't create a new one everytime.

Reply:
My bad... I read it wrong, reverted.